isobus/src/can_transport_protocol.cpp

```diff
@@ -11,860 +11,1004 @@
+		{
+			switch (message.get_identifier().get_parameter_group_number())
+			{
+				case static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolConnectionManagement):
+				{
+					if (CAN_DATA_LENGTH == message.get_data_length())
+					{
+						process_connection_management_message(message);
+					}
+					else
+					{
+						CANStackLogger::warn("[TP]: Received a Connection Management message of invalid length %hu", message.get_data_length());
+					}
+				}
+				break;
+
+				case static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolDataTransfer):
+				{
+					if (CAN_DATA_LENGTH == message.get_data_length())
+					{
+						process_data_transfer_message(message);
+					}
+					else
+					{
+						CANStackLogger::warn("[TP]: Received a Data Transfer message of invalid length %hu", message.get_data_length());
+					}
+				}
+				break;
 
-		if (nullptr != session)
-		{
-			const std::uint8_t dataBuffer[CAN_DATA_LENGTH] = { REQUEST_TO_SEND_MULTIPLEXOR,
-				                                                 static_cast<std::uint8_t>(session->get_message_data_length() & 0xFF),
-				                                                 static_cast<std::uint8_t>((session->get_message_data_length() >> 8) & 0xFF),
-				                                                 session->packetCount,
-				                                                 0xFF,
-				                                                 static_cast<std::uint8_t>(session->sessionMessage.get_identifier().get_parameter_group_number() & 0xFF),
-				                                                 static_cast<std::uint8_t>((session->sessionMessage.get_identifier().get_parameter_group_number() >> 8) & 0xFF),
-				                                                 static_cast<std::uint8_t>((session->sessionMessage.get_identifier().get_parameter_group_number() >> 16) & 0xFF) };
-			retVal = CANNetworkManager::CANNetwork.send_can_message(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolCommand),
-			                                                        dataBuffer,
-			                                                        CAN_DATA_LENGTH,
-			                                                        std::static_pointer_cast<InternalControlFunction>(session->sessionMessage.get_source_control_function()),
-			                                                        session->sessionMessage.get_destination_control_function(),
-			                                                        CANIdentifier::CANPriority::PriorityDefault6);
+				default:
+					break;
+			}
 		}
-		return retVal;
 	}
 
-	bool TransportProtocolManager::send_end_of_session_acknowledgement(TransportProtocolSession *session) const
+	void TransportProtocolManager::process_message(const CANMessage &message, void *parent)
 	{
-		bool retVal = false;
+		if (nullptr != parent)
+		{
+			reinterpret_cast<TransportProtocolManager *>(parent)->process_message(message);
```

Comment:
```suggestion
			static_cast<TransportProtocolManager *>(parent)->process_message(message);
```
I think we'd always know it'd be safe to static cast it, especially since there's no polymorphism now with this object